### PR TITLE
{chem} GAMESS-US Apr 20 2017 R1 configs, patches

### PR DESCRIPTION
**Pull request by ostueker**
Configs and patches for  GAMESS-US-20170420-R1.

To work properly they need to have this patch applied to the easyblock:
https://github.com/easybuilders/easybuild-easyblocks/pull/1221

### CODE REVIEW

File: easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1-foss-2016b.eb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+description = """ The General Atomic and Molecular Electronic Structure System (GAMESS)
+ is a general ab initio quantum chemistry package. """
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+toolchainopts = {'usempi': True}
+
+# manually download via http://www.msg.chem.iastate.edu/gamess/download.html (requires registration)
+# rename gamess-current.tar.gz by changing 'current' to the proper version
+sources = ['gamess-%(version)s.tar.gz']
+checksums = ['e9fa2e725ccbd69f4d996ab68bb65ff2']
```
Comment (boegel) [inline easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1-foss-2016b.eb]: @ostueker please use `SHA256` checksums (we're trying to make this consistent for all new easyconfig)

File: easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1-intel-2016b-sockets.eb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ is a general ab initio quantum chemistry package. """
+
+toolchain = {'name': 'intel', 'version': '2016b'}
+toolchainopts = {'usempi': False}
+ddi_comm = 'sockets'
+
+# manually download via http://www.msg.chem.iastate.edu/gamess/download.html (requires registration)
+# rename gamess-current.tar.gz by changing 'current' to the proper version
+sources = ['gamess-%(version)s.tar.gz']
+checksums = ['e9fa2e725ccbd69f4d996ab68bb65ff2']
```
Comment (boegel) [inline easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1-intel-2016b-sockets.eb]: please use SHA256 checksum

File: easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1-intel-2016b.eb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+description = """ The General Atomic and Molecular Electronic Structure System (GAMESS)
+ is a general ab initio quantum chemistry package. """
+
+toolchain = {'name': 'intel', 'version': '2016b'}
+toolchainopts = {'usempi': True}
+
+# manually download via http://www.msg.chem.iastate.edu/gamess/download.html (requires registration)
+# rename gamess-current.tar.gz by changing 'current' to the proper version
+sources = ['gamess-%(version)s.tar.gz']
+checksums = ['e9fa2e725ccbd69f4d996ab68bb65ff2']
```
Comment (boegel) [inline easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1-intel-2016b.eb]: please use SHA256 checksum

File: easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1-intel-2016b-sockets.eb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+version = '20170420-R1'
+versionsuffix = '-sockets'
+
+homepage = 'http://www.msg.chem.iastate.edu/gamess/index.html'
+description = """ The General Atomic and Molecular Electronic Structure System (GAMESS)
+ is a general ab initio quantum chemistry package. """
+
+toolchain = {'name': 'intel', 'version': '2016b'}
+toolchainopts = {'usempi': False}
+ddi_comm = 'sockets'
```
Comment (boegel) [inline easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1-intel-2016b-sockets.eb]: this should be moved down, along with the other GAMESS-US-specific parameters (`maxcpus`, etc.)

File: easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1_recent-gcc.patch [hunk 5]
```diff
@@ -0,0 +1,52 @@
+--- config.orig	2017-07-18 19:34:32.056678893 +0000
```
Comment (boegel) [inline easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1_recent-gcc.patch]: Please include a short description above this line + name of author or reference to upstream patch.

Maybe also include other GCC version number like 6.1, 6.2, 6.3, 6.4, 7.1 & 7.2 so this patch can be reused with other toolchains?

File: easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1_rungms_fix_PPN_not_initialized.patch [hunk 6]
```diff
@@ -0,0 +1,10 @@
+--- rungms.orig	2017-07-19 16:53:29.325530604 +0000
```
Comment (boegel) [inline easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1_rungms_fix_PPN_not_initialized.patch]: Please include a short description above this line + name of author or reference to upstream patch.

**Comment by boegel** [comment]
@ostueker I'm looking into testing this, but it's hanging in configure when building `GAMESS-US/20170420-R1-foss-2016b`... It seems to be waiting for input on this question (based on attaching `strace`):

```
Enter your choice of 'mkl' or 'atlas' or 'acml' or 'pgiblas' or 'none':
```

Did this work fine for you?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2420.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/0f7ec01650a9ba798daae61eaf9684b4 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2141.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/069fe14c1f6f0710b7761f2d59cf7578 for a full test report.

**Comment by ostueker** [comment]
> Did this work fine for you?

At ComputeCanada we don't have the default toolchains, therefore I can't test it with foss. :-(
However I tested successfully with our `gomkl` toolchain (GCC 5.4.0, OpenMPI 2.1.1, +MKL).
I've tried to adapt this [GAMESS-US-20170420-R1-gomkl-2016.4.11.eb][1] easyconfig to use foss instead.

Probably the question has changed slightly from the last version and needs to be tweaked in the easyblock. I'll have a look.

[1]: https://github.com/ComputeCanada/easybuild-easyconfigs/blob/computecanada-master/easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20170420-R1-gomkl-2016.4.11.eb

**Comment by boegel** [comment]
@ostueker Or maybe it doesn't deal well with not having Intel MKL available?

Since the `intel` easyconfigs work just fine, I propose that you remove the `foss` easyconfig from this pull request, and open a new separate PR for it.

That way, we can merge this one and include the easyconfigs using the `intel` toolchain in the upcoming EasyBuild v3.4.0 (last PRs are being merged in the next couple of hours for that release).

**Comment by ostueker** [comment]
I'm pretty sure the issue is here in the easyblock:

```diff
diff --git a/easybuild/easyblocks/g/gamess_us.py b/easybuild/easyblocks/g/gamess_us.py
index 11a221b..0437020 100644
--- a/easybuild/easyblocks/g/gamess_us.py
+++ b/easybuild/easyblocks/g/gamess_us.py
@@ -182,7 +182,7 @@ class EB_GAMESS_minus_US(EasyBlock):
             r"GAMESS build directory\? \[.*\] ": self.installdir,  # building in install directory
             r"Enter only the main version number, such as .*\nVersion\? ": fortran_ver,
             r".+gfortran version.\n( \n)?Please enter only the first decimal place, such as .*:": fortran_ver,
-            "Enter your choice of 'mkl' or .* 'none': ": mathlib,
+            r"Enter your choice of 'mkl' or .* 'none': ": mathlib,
         }
         run_cmd_qa(cmd, qa=qa, std_qa=stdqa, log_all=True, simple=True)
 
```
I have a commit that I can push there.

LMK if you want to postpone this for further testing.

**Comment by ostueker** [comment]
Please see if it works with this: easybuilders/easybuild-easyblocks#1239

If not I'll remove the foss easyconfig.

**Comment by boegel** [comment]
@ostueker I don't think that's the problem, making it a raw string doesn't make a difference there.

I think the problem may be that GAMESS-US doesn't know (anymore) how to deal with OpenBLAS...

Here's what I see in the EasyBuild log if I kill the running `./config` script:

```
Enter your choice of 'mkl' or 'atlas' or 'acml' or 'pgiblas' or 'none': You didn't provide a valid math library name, try again.

Linux distributions do not include a standard math library.

There are several reasonable add-on library choices,
       MKL from Intel           for 32 or 64 bit Linux (very fast)
      ACML from AMD             for 32 or 64 bit Linux (free)
     ATLAS from www.rpmfind.net for 32 or 64 bit Linux (free)
  PGI BLAS from Portland Group  for 32 or 64 bit Linux
and one very unreasonable option, namely 'none', which will use
some slow FORTRAN routines supplied with GAMESS.  Choosing 'none'
will run MP2 jobs 2x slower, or CCSD(T) jobs 5x slower.

Some typical places (but not the only ones) to find math libraries are
Type 'ls /opt/intel/mkl'                 to look for MKL
Type 'ls /opt/intel/Compiler/mkl'        to look for MKL
Type 'ls /opt/intel/composerxe/mkl'      to look for MKL
Type 'ls -d /opt/acml*'                  to look for ACML
Type 'ls -d /usr/local/acml*'            to look for ACML
Type 'ls /usr/lib64/atlas'               to look for Atlas
Type 'ls /opt/pgi/linux86-64/*/lib/*     to look for libblas.a from PGI
Type 'ls /opt/pgi/osx86-64/*/lib/*       to look for libblas.a from PGI

Enter your choice of 'mkl' or 'atlas' or 'acml' or 'pgiblas' or 'none':
```

**Comment by ostueker** [comment]
OK, I'll remove the foss easyconfig from this PR.

**Comment by boegel** [comment]
Going in, thanks @ostueker!

**Pull request merged**